### PR TITLE
chore: Upgrade ruby to 3.3.3

### PR DESCRIPTION
**Pull request by sojan-official**
- Upgrade the ruby version to 3.3.3

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # pre-build stage
-FROM ruby:3.2.2-alpine3.18 AS pre-builder
+FROM ruby:3.3.3-alpine3.18 AS pre-builder
```
Comment (vishnu-narayanan) [inline docker/Dockerfile:2]:
```suggestion
FROM ruby:3.3.3-alpine3.19 AS pre-builder
```

File: docker/Dockerfile [hunk 2]
```diff
@@ -73,7 +73,7 @@ RUN rm -rf /gems/ruby/3.2.0/cache/*.gem \
   && rm .gitignore
 
 # final build stage
-FROM ruby:3.2.2-alpine3.18
+FROM ruby:3.3.3-alpine3.18
```
Comment (vishnu-narayanan) [inline docker/Dockerfile:76]:
```suggestion
FROM ruby:3.3.3-alpine3.19
```

**Review by vishnu-narayanan — APPROVED** [review]
LGTM. We will have to change the docker base image to `alpine-3.19`

**Comment by sojan-official** [comment]
@vishnu-narayanan  made the changes. Do we need any additional testing for install/upgrade of Linux VMS?

**Comment by sojan-official** [comment]
failing on node error

Screenshot 2024-06-28 at 1 36 44 AM

**Pull request merged**